src/regmixer/model/transformer.py

```diff
@@ -154,6 +170,8 @@ def get_tokenizer_config(self, tokenizer) -> TokenizerConfig:
             raise e
 
     def get_warmup_steps(self, parameters: int) -> int:
+        if self.train_type == TrainType.anneal:
```

Comment:
Not sure if there'll ever be a need for a TrainType override other than anneal but  it might be cleaner if these overrides all lived in a dictionary e.g.
{TrainType.anneal: {warmup_steps:0,batch_size:1024,lr:6.1852e-5 etc}}